components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90

```diff
@@ -2230,163 +2221,195 @@ subroutine ma_activate_convproc_method2(     &
 !
 !-----------------------------------------------------------------------
 
-   use ppgrid, only: pver
-   use constituents, only: pcnst, cnst_name
+   use constituents, only: pcnst
    use ndrop, only: activate_modal
 
    use modal_aero_data, only:  lmassptr_amode, lmassptrcw_amode, &
-      lspectype_amode, ntot_amode, &
-      nspec_amode, ntot_amode, numptr_amode, numptrcw_amode, &
-      sigmag_amode, specdens_amode, spechygro, &
-      voltonumblo_amode, voltonumbhi_amode
+       nspec_amode, ntot_amode, numptr_amode, numptrcw_amode
 
    implicit none
 
 !-----------------------------------------------------------------------
 ! arguments  (note:  TMR = tracer mixing ratio)
-   integer, intent(in)     :: pcnst_extd
-   ! conu = tracer mixing ratios in updraft at top of this (current) level
-   !        The conu are changed by activation
-   real(r8), intent(inout) :: conu(pcnst_extd)    
-   real(r8), intent(inout) :: dconudt(pcnst_extd) ! TMR tendencies due to activation
-
-   real(r8), intent(in)    :: f_ent  ! fraction of updraft massflux that was
-                                     ! entrained across this layer == eudp/mu_p_eudp
-   real(r8), intent(in)    :: dt_u   ! lagrangian transport time (s) in the 
-                                     ! updraft at current level
-   real(r8), intent(in)    :: wup    ! mean updraft vertical velocity (m/s)
-                                     ! at current level updraft
-
-   real(r8), intent(in)    :: tair   ! Temperature in Kelvin
-   real(r8), intent(in)    :: rhoair ! air density (kg/m3)
-
-   real(r8), intent(in)    :: fracice ! Fraction of ice within the cloud
-                                     ! used as in-cloud wet removal rate
-   integer,  intent(in)    :: lun    ! logical unit for diagnostic output
-   integer,  intent(in)    :: lchnk  ! chunk index
-   integer,  intent(in)    :: i      ! column index
-   integer,  intent(in)    :: k      ! level index
+   integer, intent(in)     :: pcnst_extd        ! indices
+   ! conu = tracer mixing ratios in updraft at top of this (current) level. The conu are changed by activation
+   real(r8), intent(inout) :: conu(pcnst_extd)    ! [#/kg or kg/kg]
```

Comment:
Does this mean that `conu` refers to both number and mass mixing ratios, and that this subroutine can treat either type of quantity equivalently?

Reply:
yes... these variables have the first element (`ll == 0` in the old code) of number concentration for all aerosols and other elements (`do ll = 1, nspec_amode(n)` in the olde code) on mass concentration for each aerosol species.

Reply:
I am also making mental notes of how these appear in the code. We carry numbers and mmrs in different containers in the C++ code. We may need to restructure input to these routines or change the logic to comply with those data structures.